sphinxcontrib/needs/layout.py

```diff
@@ -887,6 +881,43 @@ def collapse_button(self, target="meta", collapsed="Show", visible="Close", init
 
         return coll_container
 
+    def permalink(self, image_url=None, image_height=None, image_width=None, text=None, prefix=""):
+        """
+        Shows a permanent link to the need.
+        Link can be a text, an image or both
+
+        :param image_url: image for an image link
+        :param image_height: None
+        :param image_width: None
+        :param text: text for a text link
+        :param prefix: Additional string infront of the string
+        :return:
+
+        Examples::
+
+            <<permalink()>>
+            <<permalink(image_url="icon:link")>>
+            <<permalink(text='¶')>>
+        """
+
+        if image_url is None and text is None:
+            image_url = "icon:share-2"
+            image_width = "17px"
+
+        config = self.app.config
+        permalink = config.needs_permalink_file
```

Comment:
I guess here is a small bug.

The default value is `/permalink.html`, which is used for the physical file location and the relative URL.
These two may not be the same, as docs may be stored under e.g. `examples-com/docs/v1.2/...`.

So the docs-root, used for the build, is not the same root, as used on the webserver.

I also think that the link to the permalink-file can not be absolute/static and can't be based on the webserver-root.
This is mainly because the weberser-root path may change or be different.

Scenario: docs get deployed twice, but were build only once:
1. example.com/latest/...
2. example.com/releases/1.1/...

So I think we need to calculate here a relative link to `permalink.html`, based on the current location of the need in the url-path.
I think something like this is already done at 1-2 places in the code and as far as I remember it was just "counting" the `/` (slahes) in need.docname and then `amount_slashes * "../" + "permalink.html"`.

Reply:
Ohh and the given path here may be also Windows-specific, so you would have backslashes in your URL.
This check may get irrelevant with the needed changes from above.

Reply:
I changed the default value. Now the permalink file option is meant to just change the file name as you suggested. This file is expected to be in the html build dir, i.e. next to index.html.

Reply:
I think this is not enough.
The permalink-link can not be static or relative to web-root, as root may change.

When I build the Sphinx-Needs docs locally, I get e.g. such locations in my browser:
``file:///home/daniel/workspace/sphinx/sphinxcontrib-needs/docs/_build/html/examples/index.html``

In this html-file, a need with a permalink is available. But the link is:
``file:///permalink.html?id=EX_REQ_5``

I came to the conclusion that we will not be able to calculate a link from any need object to the permalink page for each use case during build time.

First of all the calculated link must not contain any relative parts, as this link may not work anymore if the folder structure changes. E.g. `company.com/docs/part/chapter/../../permalink.html` will not work, if e.g. ``chapter`` folder gets renamed.

Second, we can not create an absolute link during build time, as this would mean that we know under which URL the final documentation gets deployed. And even if we know it, it would not work for local builds and server deployments with one build. Only one correct "base URL" could be set.

For me this means, we need to calculate the final permalink-url via JavaScript, as only then we are able to figure out, under which URL the docs got deployed.
Untested JS code:
```js
/* ID must get extracted via JS somehow */

$('.need-permalink').attr("href", URL("permalink.html?id=" + need_id, document.baseURI).href)
```
Then it should not matter, how often and under which domains the docs got deployed.

What do you think, makes this sense?

Reply:
> What do you think, makes this sense?

Totally. With the long break I somehow forgot about this complication ;-)
Thanks for your thorough review 👍 

I changed the computation of the link in `layout.py` to a relative link based on the need docname as you described earlier.
The web browser automatically resolves this to an absolute URL when the web page is rendered.
I tested with a local web server one time hosting _build/html and another test hosting _build.
It worked.
I also tested opening without web server directly from Windows file explorer. Here the link looked good as well but the permalink page itself does not work because of CORS error when loading needs.json.

So I guess I can get around more JS "magic" ;-)

Please test for yourself again.

I hope I understood you correctly and it is OK now.
Otherwise just tell me and I hope for your continued patience ;-)